### PR TITLE
Update readme with info from planning document

### PR DESCRIPTION
**Pull request by sydneyjodon-wk**
## Motivation
  
We need to add the information from the [planning document](https://docs.google.com/document/d/1xHjC66eUvX_SSBRXw-sVOxRzvenWfOpu8m1BQI6GmqQ/edit#) into the `analyzer_plugin` readme.
## Changes
  
Updated `readme.md`.
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: tools/analyzer_plugin/README.md [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+## Goal
+
+We want to improve the developer experience for Workiva developers writing React in Dart (and bring it closer to the React JS dev experience).
+
+This plugin has the potential to help achieve that goal by:
+* Providing static analysis of common pitfalls, with useful error messages and automatic fix suggestions.
+* Making the fluent interface syntax easier to write and deal with.
+* Helping automate common changes that can't be achieved with snippets alone (AKA "assists").
+* Encouraging best practices via lints.
+* (Potentially) Making it easier to navigate component nesting and the flow of props.
```
Comment (greglittlefield-wf) [inline tools/analyzer_plugin/README.md:27]: This is more of a LINK marketing pitch more than anything haha. Can we remove this an then follow up later with a better-crafted description?

File: tools/analyzer_plugin/README.md [hunk 2]
```diff
@@ -97,7 +138,45 @@ Congrats, you're debugging! 🎉
...
+
+#### Abandoned Feature Ideas
+
+* Outline for similar behavior to closing tags.
+    * Outline implementation is barely supported in analyzer plugin.
+    * Need to suggest closing tags API for plugin, maybe this would be a good entry point.
+* Breadcrumbs: part of the IDE plugin, not the analyzer plugin.
+    * Greg has most of a JetBrains plugin done.
+* [Prop drilling](https://marketplace.visualstudio.com/items?itemName=ReactEd.reacted#prop-drilling): isn't supported by the plugin.
+
```
Comment (greglittlefield-wf) [inline tools/analyzer_plugin/README.md:158]: I think these would be better documented as issues; can we remove this section?

File: tools/analyzer_plugin/README.md [hunk 3]
```diff
@@ -97,7 +138,45 @@ Congrats, you're debugging! 🎉
...
+* Some features might not be supported by all IDEs.
+* Workflow involves restarting the analysis server every time you make a change.
+* At times, lints do not resolve in the IDE unless you restart the analysis server.
+
+### Analyzer Limitations
+
+* Can easily go from AST to Elements but not the other way around (`computeNode` is deprecated). Most of the time should not be problematic since if you have access to an Element, you usually have access to the corresponding AST.
+    * Might be able to work around this using `NodeLocator`.
+* `toSource` should be used with caution since it's an approximation of the source.
+
```
Comment (greglittlefield-wf) [inline tools/analyzer_plugin/README.md:176]: Many of these items are either out of date, will be captured by issues, or aren't written with interneal audiences in mind. Can we remove these sections?

File: tools/analyzer_plugin/README.md [hunk 4]
```diff
@@ -1,7 +1,30 @@
...
+* __[Try it in your Package](#try-it-in-your-package)__
+* __[Repo Structure](#repo-structure)__
+* __[Local Development](#local-development)__
+    * [Setup](#setup)
+    * [Development Cycle](#development-cycle)
+    * [Design Principles & Coding Strategies](#design-principles--coding-strategies)
+    * [Debugging](#debugging-the-plugin)
+* __[Feature Ideas & Inspiration](#feature-ideas--inspiration)__
+* __[Limitations](#limitations)__
+
```
Comment (greglittlefield-wf) [inline tools/analyzer_plugin/README.md:17]: Thank you so much for adding a TOC!! 💯

**Pull request merged**